providers/terraform-cloud-scanner/src/scanner.ts

```diff
@@ -0,0 +1,165 @@
+import type { SetTargetProvidersTargetsRequestTargetsInner } from "@ctrlplane/node-sdk";
+import handlebars from "handlebars";
+import _ from "lodash";
+
+import { logger } from "@ctrlplane/logger";
+
+import type { Variable, Workspace } from "./types.js";
+import { listVariables, listWorkspaces } from "./api.js";
+import { env } from "./config.js";
+import { api } from "./sdk.js";
+
+const workspaceTemplate = handlebars.compile(
+  env.CTRLPLANE_WORKSPACE_TARGET_NAME,
+);
+
+/**
+ * Scans Terraform Cloud workspaces and registers them as targets with prefixed labels and a link.
+ */
+export async function scan() {
+  logger.info("Starting Terraform Cloud scan");
+
+  try {
+    const workspaces: Workspace[] = await listWorkspaces();
+    logger.info(`Found ${workspaces.length} workspaces`);
+
+    const targets: SetTargetProvidersTargetsRequestTargetsInner[] = [];
+
+    for (const workspace of workspaces) {
+      logger.info(
+        `Processing workspace: ${workspace.attributes.name} (ID: ${workspace.id})`,
+      );
+
+      const variables: Variable[] = await listVariables(workspace.id);
+      logger.info(
+        `  Found ${variables.length} variables in workspace '${workspace.attributes.name}'`,
+      );
+      const variableLabels = processVariables(variables);
+      const tagLabels = processWorkspaceTags(workspace.attributes["tag-names"]);
+      const vcsRepoLabels = processVcsRepo(workspace.attributes["vcs-repo"]);
+      const link = buildWorkspaceLink(workspace);
+      const targetName = workspaceTemplate({ workspace });
+
+      const target: SetTargetProvidersTargetsRequestTargetsInner = {
+        version: "terraform/v1",
+        kind: "Workspace",
+        name: targetName,
+        identifier: workspace.id,
+        config: {
+          workspaceId: workspace.id,
```

Comment:
should be anything else here? what information might be useful to have access to?

Reply:
Linked in discord.